src/commonMain/kotlin/fr/acinq/lightning/channel/states/Channel.kt

```diff
@@ -78,13 +78,29 @@ sealed class ChannelState {
             is Syncing -> this@ChannelState.state
             else -> this@ChannelState
         }
+        maybeSignalSensitiveTask(oldState, newState)
         return when {
             // we only want to fire the PaymentSent event when we transition to Closing for the first time
             oldState is ChannelStateWithCommitments && oldState !is Closing && newState is Closing -> emitClosingEvents(oldState, newState)
             else -> emptyList()
         }
     }
 
+    /** Some transitions imply that we are in the middle of tasks that may require some time. */
+    private fun ChannelContext.maybeSignalSensitiveTask(oldState: ChannelState, newState: ChannelState) {
```

Comment:
Don't you want to also handle initial channel creation here? Or maybe we'll do that in a separate, follow-up PR? I think this mechanism would still work, we should be able to handle those transitions here.